core-java-collections/src/test/java/com/baeldung/enummap/EnumMapUnitTest.java

```diff
@@ -0,0 +1,137 @@
+package com.baeldung.enummap;
+
+import org.junit.Test;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static java.util.AbstractMap.SimpleEntry;
+
+public class EnumMapUnitTest {
+    public enum DayOfWeek {
+        MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY
+    }
+
+    @Test
+    public void whenContructedWithEnumType_ThenOnlyAcceptThatAsKey() {
+        EnumMap<DayOfWeek, String> dayMap = new EnumMap<>(DayOfWeek.class);
+        Map genericMap = dayMap;
+        assertThatCode(() -> genericMap.put(TimeUnit.NANOSECONDS, "NANOSECONDS")).isInstanceOf(ClassCastException.class);
+    }
+
+    @Test
+    public void whenConstructedWithEnumMap_ThenSameKeyTypeAndInitialMappings() {
+        EnumMap<DayOfWeek, String> dayMap = new EnumMap<>(DayOfWeek.class);
+        dayMap.put(DayOfWeek.MONDAY, "Soccer");
+        dayMap.put(DayOfWeek.TUESDAY, "Basketball");
+
+        EnumMap<DayOfWeek, String> activityMapCopy = new EnumMap<>(dayMap);
+        assertThat(activityMapCopy.size()).isEqualTo(2);
+        assertThat(activityMapCopy.get(DayOfWeek.MONDAY)).isEqualTo("Soccer");
+        assertThat(activityMapCopy.get(DayOfWeek.TUESDAY)).isEqualTo("Basketball");
+    }
+
+    @Test
+    public void givenEmptyMap_whenConstructedWithMap_ThenException() {
+        HashMap ordinaryMap = new HashMap();
+        assertThatCode(() -> new EnumMap(ordinaryMap)).isInstanceOf(IllegalArgumentException.class).hasMessage("Specified map is empty");
```

Comment:
This line scrolls horizontally. Could we please break it up to make it easier for our readers?

```suggestion
        assertThatCode(() -> new EnumMap(ordinaryMap))
          .isInstanceOf(IllegalArgumentException.class)
          .hasMessage("Specified map is empty");
```